src/particles/spacecharge/EnvelopeSpaceChargePush.cpp

```diff
@@ -0,0 +1,68 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Marco Garten, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#include "EnvelopeSpaceChargePush.H"
+
+#include <AMReX_BLProfiler.H>
+#include <AMReX_REAL.H>       // for Real
+#include <AMReX_SmallMatrix.H>
+
+#include <cmath>
+
+namespace impactx::spacecharge
+{
+        /** This function returns the linear transport map associated with a
+         *  reduced 2D space charge model, based on the beam covariance matrix.
+         *
+         * @param[in] refpart reference particle
+         * @param[in,out] cm covariance matrix
+         * @param[in] current beam current [A]
+         * @param[in] ds step size [m]
+         * @returns 6x6 transport matrix
+         */
+        AMREX_GPU_HOST AMREX_FORCE_INLINE
+        void
+        envelope_space_charge2D_push (
+        RefPart const & refpart,
+        Map6x6 & cm,
+        amrex::ParticleReal & current,
+        amrex::ParticleReal & ds
```

Comment:
See interface file comment:
```suggestion
        amrex::ParticleReal current,
        amrex::ParticleReal ds
```
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in